### ISSUE TITLE
Vector definition through sympify method

### ISSUE THREAD

**Issue by LadonkinV** [sympy/sympy#17978]
Seems that Vector object defined through sympify method contains issue. The sympify function performed well, but output of the result Vector object shows an AttributeError, as well any attempt to conduct math operation with this Vector object leads to endless loop(memory overload) or error.

```
from sympy.vector import *
from sympy import *
expr = sympify('Vector(R.i*3, R.j*4, R.k*5)', locals=dict(R=CoordSys3D('R'), Vector=Vector))
expr
```

```
...
/srv/conda/lib/python3.6/site-packages/sympy/vector/vector.py in components(self)
     39         # The '_components' attribute is defined according to the
     40         # subclass of Vector the instance belongs to.
---> 41         return self._components
     42 
     43     def magnitude(self):

AttributeError: 'Vector' object has no attribute '_components'
```

**Comment by friyaz** [comment]
I think the error is due to creating an object of Vector class, not due to simpify. Objects of Vector class should not be created directly rather base scalars and vectors be used.

You can try:
```python
from sympy.vector import *
from sympy import *
expr = sympify('R.i*3 + R.j*4 + R.k*5', locals=dict(R=CoordSys3D('R'), Vector=Vector))
expr.components
{R.j: 4, R.i: 3, R.k: 5}
```

**Comment by oscarbenjamin** [comment]
I think it would be better if Vector gave a better error message since it isn't supposed to be constructed that way and most people would find that a more intuitive way of constructing a vector.

I also think it would be better for most simple uses if Vector had a "standard" coordinate system that could be used without needing to first create a CoordSys3D. For example you could have
```
from sympy.vector.stdcoord import i, j, k

v = 2*i + 3*j + 4*k
```
Ideally the standard coordinate system should display just as `2*i` rather than `2*N.i`. If we have that then there could be a function to create a vector in the standard coordinate system like `vector_std(x, y, z)`.

It's good that the Vector module has the capability for multiple coordinate systems built into it but it's also overkill for most use cases. I suspect that the vast majority of users only need one coordinate system so we should make that case as easy as possible. Probably a lot of the time someone will just want to calculate a cross product or something.

**Comment by friyaz** [comment]
I agree. I also find using base scalars and base vector with `C.x` and `C.i` non-intuitive. It would be nice if we can just use `i`, `j`, and `k` for base vectors and `x`, '`y`, and '`z' for base scalars.

If everyone is ok with this, I will try to make a PR for this.

**Comment by meganly** [comment]
What's the status of this issue?

**Comment by oscarbenjamin** [comment]
I don't think anything has happened since the comments above. I would definitely be in favour of introducing a standard coordinate system and also writing a new basic docs page that just shows how to do common vector calculations in the standard coordinate system.

**Comment by meganly** [comment]
I would as well though I'm not sure I quite understand how this would be implemented.  Should the standard coordinate system be a subclass of CoordSys3D?

**Comment by oscarbenjamin** [comment]
I think it should be an instance of `CoordSys3D` although perhaps to make the printing different it would need to be an instance of a subclass.

**Comment by oscarbenjamin** [comment]
The easiest thing to being would just be to create an instance of `CoordSys3D`

**Comment by meganly** [comment]
The printing of basis vectors is hard-coded so it would need to be a subclass.  I stumbled upon this issue after getting help on Gitter about how to print the latex for the standard basis vectors as '\mathbf{i}', '\mathbf{j}', '\mathbf{k}'.

I don't like how the base vectors are initialized by defining the printers on the objects, but the following solved my printing issue.

```python
class StdCoord(CoordSys3D):
    def __init__(self, *args, **kwargs):
        latex_vects = [r'\mathbf{%s}' % x for x in self._vector_names]
        pretty_vects = ['%s' % x for x in self._vector_names]

        v1 = BaseVector(0, self, pretty_vects[0], latex_vects[0])
        v2 = BaseVector(1, self, pretty_vects[1], latex_vects[1])
        v3 = BaseVector(2, self, pretty_vects[2], latex_vects[2])

        self._base_vectors = (v1, v2, v3)

        setattr(self, self._vector_names[0], v1)
        setattr(self, self._vector_names[1], v2)
        setattr(self, self._vector_names[2], v3)
```

The basis vectors i, j, and k are attributes of the coordinate system so I don't think this is quite you have in mind.
```python
>>> R = StdCoord('R')
>>> i, j, k = R.base_vectors()
>>> i
R.i
```

**Comment by oscarbenjamin** [comment]
You can do `R.i._name = 'i'`.

It's all a bit messy but it should be possible to change `CoordSys3D.__new__` and/or `BaseVector.__new__` to make this work more cleanly.

**Comment by meganly** [comment]
Yes, you can make this work by doing `obj._name = name` in `BaseVector.__new__`.  But then that changes how base vectors get printed in `CoordSys3D`.

I don't see how to get vectors to print one way in `CoordSys3D` and another way in `StdCoord` without subclassing `CoordSys3D` and `BaseVector`.  This feels overly complicated and doesn't seem to be the right way to go about implementing `StdCoord`.

**Comment by oscarbenjamin** [comment]
I think the right way is to make some changes to `CoordSys3D` and `BaseVector` so that it is straight-forward to customise the printing in either an instance or a subclass.

**Comment by meganly** [comment]
Makes sense.  On Gitter @asmeurer told me 

>I don't like how the vector module defines the printers on the objects. SymPy library classes should define printing on the printer classes. I also don't like how it precomputes the latex form at construction time. 

Does this mean `BaseVector` handles printing by defining some ` _latex(self, printer)` function?

**Comment by oscarbenjamin** [comment]
I think it's the `_sympystr` method and possibly the `_latex_form` attribute.

**Comment by meganly** [comment]
Modifying the `_sympystr` method and `latex_form` attribute will change the printing, but that still defines printers on the objects and precomputes the latex form at construction time.

**Comment by meganly** [comment]
What about passing in a keyword, like `abbrev_names`,  to `CoordSys3D` and `BaseVector` that if set to True doesn't print references to the coordinate system name?

Then in `CoordSys3D` we could do
```python
if abbrev_names:
  latex_scalars = [(r"\mathbf{{%s}}" % x) for x in variable_names]
else:
  latex_scalars = [(r"\mathbf{{%s}_{%s}}" % (x, name)) for x in variable_names]
```
and in `BaseVector`
```python
if abbrev_names:
  obj._name = name
else:
  obj._name = system._name + '.' + name
```
This would give
```python
>>> R = StdCoord('R', abbrev_names=True)
>>> R.i
i
>>> latex(R.i)
'\\mathbf{\\hat{i}}'
```
